blocks/search.py

```diff
@@ -277,6 +276,7 @@ def search(self, input_values, eol_symbol, max_length,
             negative log-likelihood.
 
         """
+        self.beam_size = input_values[input_values.keys()[0]].shape[1]
```

Comment:
It's better to provide beam size as an additional argument to `compute_initial_states`, anyway you don't use it anywhere else.